### PR TITLE
Feature/integration test

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -112,3 +131,32 @@ tasks.docker.dependsOn tasks.build
...
+        java {
+            compileClasspath += main.output + test.output
+            runtimeClasspath += main.output + test.output
+            srcDir 'src/itest/java'
+        }
+        resources.srcDir 'src/itest/resources'
+    }
+}
+
+task itestSetup(type: Exec) {
```
Comment (mcornaton) [inline build.gradle:146]: Nothing to do in `itestSetup`/`itestTeardown` ?

Comment (jeremyjams) [reply build.gradle:146]: No yet indeed :)

Comment (jeremyjams) [reply build.gradle]: That was on purpose in order to keep same structure than other projects but we can remove it yes, done :+1:

File: src/itest/java/com/iexec/worker/tee/post/compute/IntegrationTests.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    public static final String MOCK_WORKER_DOCKER_COMPOSE_YML =
+            RESOURCES_DIR + "/docker-compose.yml";
+    public static final String CONTAINERS_PREFIX = "iexec-";
+    public static final String WORKER = "worker";
+    public static final String POST_COMPUTE = "post-compute";
+    public static final String INTERNAL_NETWORK = "iexec-post-compute-net";
+    public static final String TEE_WORKER_POST_COMPUTE_IMAGE =
+            "nexus.iex.ec/tee-worker-post-compute:dev";
+
+    @ClassRule
```
Comment (zguesmi) [inline src/itest/java/com/iexec/worker/tee/post/compute/IntegrationTests.java:49]: Is this still functional with Junit 5? Shouldn't it be `@Extension`?

File: src/itest/resources/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+version: '3.5'
+
+networks:
+  iexec-post-compute-net:
+    name: iexec-post-compute-net
+    external: false
+
+services:
+  worker:
+    image: wiremock/wiremock
```
Comment (zguesmi) [inline src/itest/resources/docker-compose.yml:10]: Nice!

File: src/itest/java/com/iexec/worker/tee/post/compute/IntegrationTests.java (hunk 2 above)
Comment (jbern0rd) [reply src/itest/java/com/iexec/worker/tee/post/compute/IntegrationTests.java:49]: It is not clearly documented on testcontainers.org, but `@ClassRule` annotation is for JUnit4 as mentioned by @zguesmi . (https://www.testcontainers.org/quickstart/junit_4_quickstart/)
With JUnit5, you have to use the `@Container` annotation for single containers as well as for docker compose.
```suggestion
    @Container
```

Comment (jbern0rd) [reply src/itest/java/com/iexec/worker/tee/post/compute/IntegrationTests.java:49]: https://www.testcontainers.org/quickstart/junit_5_quickstart/

**Review by jbern0rd — APPROVED** [review]
Thanks

**Review by mcornaton — APPROVED** [review]

**Review by zguesmi — APPROVED** [review]
Thanks for the updates!

**Pull request merged**